projects/step-core/src/lib/components/dynamic-textfield/dynamic-textfield.component.ts

```diff
@@ -0,0 +1,96 @@
+import { Component, forwardRef, Input, ViewEncapsulation } from '@angular/core';
+import { NG_VALUE_ACCESSOR } from '@angular/forms';
+import { DynamicValueString } from '../../client/step-client-module';
+import { DialogsService } from '../../shared';
+
+type OnChange = (dynamicValueString: DynamicValueString) => void;
+type OnTouch = () => void;
+
+@Component({
+  selector: 'step-dynamic-textfield',
+  templateUrl: './dynamic-textfield.component.html',
+  styleUrls: ['./dynamic-textfield.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => DynamicTextfieldComponent),
+      multi: true,
+    },
+  ],
+})
+export class DynamicTextfieldComponent {
+  @Input() label?: string;
+  @Input() tooltip?: string;
+  @Input() invalid: boolean = false;
+  @Input() required: boolean = false;
```

Comment:
If component implements `ControlValueAccessor` it's validity state should be determined by `ng-invalid` class, that is automatically set by assigned Control in case of it's invalidity.

At the same time Control's validity/invalidity behavior should be setup by form's validators and not by additional props.